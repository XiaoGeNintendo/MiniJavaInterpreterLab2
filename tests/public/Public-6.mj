
// explicit constructor
// with inheretance

class A {
    int a;

    A() {
        a = 100;
    }
}

class B extends A {
    string b;

    B() {
        this(10, "hello");
    }

    B(int a, string b) {
        super();

        println(this.a);

        this.a = a;
        this.b = b;
    }
}

int main() {    
    B x1 = new B();
    println(x1.a);
    println(x1.b);

    return 0;
}