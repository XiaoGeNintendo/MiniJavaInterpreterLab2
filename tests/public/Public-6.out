100
10
hello
Process exits with the code 0.
